### PR TITLE
[JSON RPC] add MockLibraDB for testing

### PR DESCRIPTION
**Pull request by christinacelee**
## Motivation

Adding `MockLibraDB` for unit-testing JSON RPC 
Added new trait `LibraDBTrait` that both `LibraDB` and `MockLibraDB` implement

## Test Plan

Refactored tests pass

### CODE REVIEW

File: json-rpc/src/mock_db.rs [hunk 1]
```diff
@@ -0,0 +1,36 @@
+// Copyright (c) The Libra Core Contributors
```
Comment (phoenix-antigravity) [inline json-rpc/src/mock_db.rs]: nit: make separate folder for tests and move mock_db and existing coverage there for better isolation

**Comment by thefallentree** [comment]
This means we need to reimplement all the logics at MockLibraDB layer, that seems to be very error prone (we are not actually using tree structure for example, and maybe this should be part of proptest instead). I thought we should mock out SchemaDB  layer,  we simply don't need to use rocksdb (use a in-memory k/v is more than enough) , but will keep most of the LibraDB logic intact.

**Comment by christinacelee** [comment]
I think for unit-testing purposes we don't need to mock to that level.. the unit tests for JSON RPC calls should just test the JSON RPC part (client makes a JSON RPC call, and test whether the expected value (whatever JSON RPC grabs from DB, mocked or not) matches the actual value returned from JSON RPC calls. We shouldn't have to worry about the integrity of the DB itself here - there are tests already for that in `storage`. Even the existing `MockStorageReadClient` mocks the payload/data itself

**Pull request closed without merging**